arrows/qt/widgets/LoggerView.ui

```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>LoggerView</class>
+ <widget class="QWidget" name="LoggerView">
+  <property name="geometry">
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>284</width>
+    <height>220</height>
+   </rect>
+  </property>
+  <layout class="QGridLayout" name="gridLayout_2">
```

Comment:
Why are there nested layouts here? The outer one doesn't seem to have an obvious purpose?

I'd also probably use a box layout instead.

Reply:
In fact, I don't think I'd use a `.ui` at all. Not for a single widget.

I'd *almost* be tempted to just have `LoggerViewer` subclass `QPlainTextEdit` directly, but that has all sorts of potential encapsulation drawbacks.

Reply:
```c++
  auto* const layout = new QVBoxLayout{this};
  layout->setContentsMargins(0, 0, 0, 0);
  layout->addWidget(d->m_foo = new QPlainTextEdit{this});

  d->m_foo->setLineWrapMode(QPlainTextEdit::NoWrap);
  d->m_foo->setReadOnly(true);
```

(Obviously use something other than `m_foo`, I just don't have a strong feeling what that should be. `m_log`? `m_textEdit`? Something else?)

Reply:
This is somewhat of a placeholder.  It's a single widget for now, but I plan to add more bells and whistles.  For example, filters to show logs from only some sources, possibly some formatting options, etc.  So I think in the long term it make sense for this to be a `.ui`.

This was my first attempt at doing layouts, so I'm sure it's not ideal.  I can play with it a little.

Reply:
I don't know how to do the layouts in Designer.  If you want to propose something better that's fine, but I'm not going to take the time to figure it out.  This works for now.